src/field/arena.ts

```diff
@@ -71,6 +71,44 @@ export class Arena {
     this.updatePoolsForTimeOfDay();
   }
 
+  /**
+   * Determines if the arena is of the specified biome
+   * @param biome - {@linkcode Biome} or array of {@linkcode Biome} to check against
+   * @returns `true` if the arena is of the specified biome, `false` otherwise
+   */
+  public isInBiome(biome: Biome | Biome[]): boolean {
+    return Array.isArray(biome) ? biome.includes(this.biomeType) : this.biomeType === biome;
+  }
+
+  /**
+   * Determines if the arena is of the specified terrain
+   * @param terrain - {@linkcode TerrainType} or array of {@linkcode TerrainType} to check against
+   * @returns `true` if the arena is of the specified terrain, `false` otherwise
+   */
+  public hasTerrain(terrain: TerrainType | TerrainType[]): boolean {
+    const terrainType = this.getTerrainType();
+    return Array.isArray(terrain) ? terrain.includes(terrainType) : terrainType === terrain;
+  }
+
+  /**
+   * Determines if the arena is of the specified weather
```

Comment:
```suggestion
   * Determines if one the specified weather effects is set in the arena.
   * Does **not** take in account weather suppression effects.
```
I think that's true, right?
Would also be nice to {@linkcode} to what function to call if we need to check for suppressed weather?